src/contentBase.ts

```diff
@@ -88,23 +88,44 @@ export abstract class ContentBase implements IPackageMetadata {
             return null;
         }
         searchFile = searchFile.toLowerCase();
+        let foundPath = null;
         for(const filePath of fileList) {
             if(filePath.toLowerCase() === searchFile) {
                 return filePath;
             }
-            const pathSplit =  filePath.split(path.sep);
+            const pathSplit = filePath.split(path.sep);
             for(const pathSegment of pathSplit) {
                 if(pathSegment.toLowerCase() === searchFile && (searchFile !== Constants.INFO_PLIST || searchFile === Constants.PROVISIONING)) {
-                    return filePath;
+                    foundPath = this.checkUpdatePath(foundPath, filePath);
                 }
                 // Info.plist and embedded.mobileprovisions is a special case since there are multiple instances in different files
                 if(pathSegment.toLowerCase() === searchFile && (searchFile === Constants.INFO_PLIST || searchFile === Constants.PROVISIONING) && pathSplit.length <= 3) {
-                    return filePath;
+                    foundPath = this.checkUpdatePath(foundPath, filePath);
                 }
             }
         }
-        return null;
+        return foundPath;
     }
+
+
+    /**
+     * This method determines which provision file to be selected when there is more than one. 
```

Comment:
This method is agnostic to the type of file being used. Reflect that in the comment. (same with the param names)